addons/stock/report/report_lot_barcode.xml

```diff
@@ -25,7 +25,8 @@
                                         <t name="gs1_datamatrix_lot" t-if="o.product_id.tracking == 'lot'" t-set="final_barcode" t-value="(final_barcode or '') + '10' + o.name"/>
                                         <t t-elif="o.product_id.tracking == 'serial'" t-set="final_barcode" t-value="(final_barcode or '') + '21' + o.name"/>
                                     </t>
-                                    <div class="o_label_4x12" t-field="o.product_id.display_name" t-att-style="'width:22mm' if final_barcode else ''"/>
+                                    <div class="o_label_4x12" t-att-style="'width:22mm' if final_barcode else ''">[<span t-field="o.product_id.default_code"/>]</div>
```

Comment:
```suggestion
                                    <div class="o_label_4x12" t-att-style="'width:22mm' if final_barcode else ''">[<t t-out="o.product_id.default_code"/>]</div>
```
Having a `<span>` here creates an unwanted padding.

Also, this raises issues with the display of a classic barcode:
![image](https://github.com/odoo/odoo/assets/18055894/d3da379e-acc2-44fd-845f-5130f076b3de)

Reply:
Changed `span` to `t` :+1: